### PR TITLE
[PHPStan] Fix level 2 errors

### PR DESCRIPTION
**Pull request by blankse**
Before
`[ERROR] Found 610 errors`

After
`[ERROR] Found 477 errors`

### CODE REVIEW

**Comment by brusch** [comment]
@blankse Ready for review or still in progress?

**Comment by blankse** [comment]
@brusch I think it's enough :) You can review it.

File: composer.json [hunk 1]
```diff
@@ -101,12 +101,23 @@
     "cache/integration-tests": "^0.16.0",
     "codeception/codeception": "~2.4.5",
     "phpstan/phpstan": "^0.12",
-    "phpstan/phpstan-symfony": "^0.12"
+    "phpstan/phpstan-symfony": "^0.12",
+    "heidelpay/heidelpay-php": "^1.2.5.1",
```
Comment (brusch) [inline composer.json]: Imho those dependencies should stay optional.

File: lib/Model/ModelInterface.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     * @throws \Exception
+     */
+    public function initDao($key = null, $forceDetection = false);
+
+    /**
+     * @param string $modelClass
+     *
+     * @return string|null
+     */
+    public static function locateDaoClass($modelClass);
```
Comment (brusch) [inline lib/Model/ModelInterface.php]: Not really needed on the interface, right?

File: composer.json (hunk 1 above)
Comment (blankse) [reply composer.json]: Its only "require-dev". Is only installed if the composer is run directly on the package (root-only). See https://getcomposer.org/doc/04-schema.md#require-dev

Comment (brusch) [reply composer.json]: Oh, sorry, missed that detail in the diff view 😊 Thanks for clarifying!

Comment (blankse) [reply composer.json]: I added this packages also to `suggest`. So the user see, which packages can be added to increase functionality.

File: lib/Model/ModelInterface.php (hunk 2 above)
Comment (blankse) [reply lib/Model/ModelInterface.php]: You are right. I removed it.

File: bundles/AdminBundle/Controller/Admin/Document/FolderController.php [hunk 3]
```diff
@@ -42,6 +42,10 @@ public function getDataByIdAction(Request $request)
     {
         $folder = Document\Folder::getById($request->get('id'));
 
+        if (!$folder) {
+            throw $this->createNotFoundException();
```
Comment (weisswurstkanone) [inline bundles/AdminBundle/Controller/Admin/Document/FolderController.php]: maybe add a message

Comment (blankse) [reply bundles/AdminBundle/Controller/Admin/Document/FolderController.php]: Default message is "Not found" and its a 404. Not enough?

Comment (weisswurstkanone) [reply bundles/AdminBundle/Controller/Admin/Document/FolderController.php]: for now I would leave it as it is

Comment (blankse) [reply bundles/AdminBundle/Controller/Admin/Document/FolderController.php]: I set it, so the user know at least the type

**Comment by weisswurstkanone** [comment]
thanks a lot!

**Pull request merged**